packages/core/src/request.ts

```diff
@@ -94,7 +95,7 @@ export class Request {
     headers?: Record<string, string>;
 
     /** Custom user data assigned to the request. */
-    userData?: Record<string, unknown>;
+    userData: Record<string, unknown> & { '__crawlee'?: Record<string, unknown> };
```

Comment:
would be better to keep the old type and use casting where needed instead of publicly mentioning the `__crawlee` property, given we want users to not know about this implementation detail

i would be fine with `Record<string, any>` too, which won't require any casting